microindex/finder.go

```diff
@@ -40,27 +40,63 @@ func NewFinder(ctx context.Context, zctx *resolver.Context, uri iosrc.URI) (*Fin
 }
 
 // lookup searches for a match of the given key compared to the
-// key values in the records read from the reader.  If the boolean argument
-// "exact" is true, then only exact matches are returned.  Otherwise, the
-// record with the lagest key smaller than the key argument is returned.
-func lookup(reader zbuf.Reader, compare expr.KeyCompareFn, exact bool) (*zng.Record, error) {
+// key values in the records read from the reader.  If the op argument is "="
+// then only exact matches are returned.  Otherwise, the record with the
+// largest key smaller (or larger) than the key argument is returned.
+func lookup(reader zbuf.Reader, compare expr.KeyCompareFn, dir zbuf.Order, op string) (*zng.Record, error) {
```

Comment:
Maybe define integer constants (like eq, gte, and lte) for op instead of using a string?

Reply:
Yeah probably should.